### PR TITLE
Change the default arg parser to typer

### PR DESCRIPTION
**Pull request by jhonnye0**
## Pull Request description
Change the default arg parser to typer

## How to test these changes

* Run smoke and unit tests to see if there's no errors
* use the new cli sugar --help

* ```...```

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [x] new feature
- [ ] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [ ] pre-commit hooks were executed locally.
- [x] this PR requires a project documentation update.

Author's checklist:
- [x] I have reviewed the changes and it contains no misspelling.
- [x] The code is well commented, especially in the parts that contain more complexity.
- [x] New and old tests passed locally.

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

### CODE REVIEW

**Comment by xmnlab** [comment]
@jhonnye0 , thanks for working on that!

the CI is failing here: https://github.com/osl-incubator/sugar/actions/runs/8933790095/job/24539743195?pr=117#step:5:10

you need to remove the poetry.lock and generate it again with `poetry lock`
maybe it is not necessary to remove .. as it will generate it again .. but just in case I remove it to be sure about the result

File: src/sugar/cli.py [hunk 1]
```diff
@@ -1,104 +1,26 @@
 """Definition of the CLI structure."""
-import argparse
+from __future__ import annotations
+
 import os
 import sys
 
 from pathlib import Path
-from typing import Tuple
+from typing import Any, List, Tuple
```
Comment (xmnlab) [inline src/sugar/cli.py:8]: at the beginning you have `from __future__ import annotations`

so here you don't need to import `List` anymore.
in you code you can annotate the type with `list`, for example: `list[str]`

`List` is deprecated

File: src/sugar/cli.py [hunk 2]
```diff
@@ -155,13 +77,1672 @@ def extract_options_and_cmd_args() -> Tuple[list, list]:
...
+        cmd_args: List[Any] = state['cmd']
+        opts_args: List[Any] = state['options']
+
+        Sugar(args, options_args=opts_args, cmd_args=cmd_args).run()
+
+    # -- Stats Commands --
+
+    @stats_group.command()
+    def plot():
+        print('Plot command')
```
Comment (xmnlab) [inline src/sugar/cli.py:1740]: it seems that this function is incomplete.

**Review by xmnlab — CHANGES_REQUESTED** [review]
@jhonnye0 in general looks really good. thanks for working on that.
maybe as a follow-up it would be nice to create these commands dynamically from some kind of schema

File: src/sugar/cli.py (hunk 1 above)
Comment (jhonnye0) [reply src/sugar/cli.py:8]: Oh sure, I'll change that

File: src/sugar/cli.py (hunk 2 above)
Comment (jhonnye0) [reply src/sugar/cli.py:1740]: Sorry haha, I forgot this one xD

**Comment by xmnlab** [comment]
@jhonnye0 it seems that it has some issues with poetry ... 
probably the best way to fix that is to remove poetry.lock and run `poetry lock` again

**Review by xmnlab — APPROVED** [review]
LGTM! thanks @jhonnye0 !

it would be nice to have a follow up to make it create the typer commands dynamically from schema ... let me know if you would like on that as well :)

it is really a great new feature for the project :)

appreciate that.

**Pull request merged**